### PR TITLE
Fall cleanup

### PR DESCRIPTION
**Pull request by niik**
Time to give some maintenance love to dugite.

- Upgrades typescript to 4.7
- Enforces Node12+
- Runs CI on Node 12, 14, 16
- Switches us to using yarn (closes: #277)
- Removes `got` dependency in favor of built-in https module
- Removes `checksum` dependency in favor of built-in crypto module
- Removed `rimraf` dependency in favor of built-in recursive remove

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -30,23 +36,23 @@ jobs:
       - uses: actions/checkout@v2
         with:
           submodules: recursive
-      - name: Install Node.js ${{ matrix.arch }}
-        uses: niik/setup-node@f9547c97f4c519f71dc42e652d6d2c53f9527c1d
```
Comment (sergiou87) [inline .github/workflows/ci.yml:34]: Didn't notice we used a custom `setup-node` 🤯

File: docs/releases.md [hunk 2]
```diff
@@ -39,12 +41,13 @@ After that, don't forget publish the release on the repo.
...
-# to ensure everything is up-to-date and tests pass
-npm ci
+# to ensure everything is up-to-date
+yarn
 
-# if you have not run `npm run build` before, a couple of you cloning auth test will fail.
-npm test 
+yarn build
 
+# if you have not run `npm run build` before, a couple of you cloning auth test will fail.
```
Comment (sergiou87) [inline docs/releases.md:49]: I guess we should update this comment too:
```suggestion
# if you have not run `yarn build` before, a couple of you cloning auth test will fail.
```

File: test/slow/git-spawn-test.ts [hunk 3]
```diff
@@ -12,13 +12,15 @@ const maximumStringSize = 268435441
...
-    })
+    if (process.stdout) {
+      process.stdout.on('data', chunk => {
+        if (chunk instanceof Buffer) {
+          stdout.push(chunk)
+        } else {
+          stdout.push(Buffer.from(chunk))
+        }
+      })
+    }
```
Comment (sergiou87) [inline test/slow/git-spawn-test.ts:23]: Can't we use optional chaining here? 🤔 
```suggestion
    process.stdout?.on('data', chunk => {
      if (chunk instanceof Buffer) {
        stdout.push(chunk)
      } else {
        stdout.push(Buffer.from(chunk))
      }
    })
```

**Review by sergiou87 — CHANGES_REQUESTED** [review]
Great cleanup! 🤩  Only a couple of comments/suggestions

File: .github/workflows/ci.yml (hunk 1 above)
Comment (niik) [reply .github/workflows/ci.yml:34]: Yeah, and from some super untrustworthy source too!

File: test/slow/git-spawn-test.ts (hunk 3 above)
Comment (niik) [reply test/slow/git-spawn-test.ts:23]: Not yet, the version of prettier that we use in dugite didn't support which is why I did it this way (sorry, should have made a comment or at least a review comment about it) it but I'm planning on bumping prettier in a separate PR (this one got noisy enough)

**Comment by niik** [comment]
@sergiou87 Would you like me to do a prettier PR targeting this or are you happy to approve this as-is?

**Review by sergiou87 — APPROVED** [review]
Happy to approve it as is! (sorry, I missed your changes after my review two weeks ago 😓 )

**Comment by niik** [comment]
> sorry, I missed your changes after my review two weeks ago

Oh no worries, I got distracted away from it anyway. I should have requested a re-review. Thanks for taking a look!

**Pull request merged**